app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -97,19 +110,81 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
         }
 
         // Sync stopped successfully.
-        override fun onIdle() {
+        override fun onIdle(result: SyncResult) {
             CoroutineScope(Dispatchers.Main).launch {
                 val pref = findPreference<Preference>(context!!.getPreferenceKey(R.string.pref_key_sync_now))
                 pref?.let {
                     pref.title = getString(R.string.preferences_sync_now)
                     pref.isEnabled = true
                     updateLastSyncedTimePref(context!!, pref, failed = false)
                 }
+
+                Log.i(LOGTAG, "Sync finished; recording telemetry")
+                for ((_, status) in result) {
```

Comment:
I'd personally move this code somewhere in the a-c component, as this doesn't need to be app specific (I think).